test/Lint.lean

```diff
@@ -116,3 +116,93 @@ info: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
+which would be better stated with explicit decidability statements.
+Instead, use `open Classical in` for definitions or instances, the `classical` tactic for proofs.
+For theorem statements, either add missing decidability assumptions or use `open Classical in`.
+note: this linter can be disabled with `set_option linter.openClassical false`
+-/
+#guard_msgs in
+open Nat Classical Nat
+
+/--
+warning: please avoid 'open (scoped) Classical' statements: this can hide theorem statements
+which would be better stated with explicit decidability statements.
+Instead, use `open Classical in` for definitions or instances, the `classical` tactic for proofs.
+For theorem statements, either add missing decidability assumptions or use `open Classical in`.
+note: this linter can be disabled with `set_option linter.openClassical false`
+-/
+#guard_msgs in
+open Classical hiding choose
+
+/--
+warning: please avoid 'open (scoped) Classical' statements: this can hide theorem statements
+which would be better stated with explicit decidability statements.
+Instead, use `open Classical in` for definitions or instances, the `classical` tactic for proofs.
+For theorem statements, either add missing decidability assumptions or use `open Classical in`.
+note: this linter can be disabled with `set_option linter.openClassical false`
+-/
+#guard_msgs in
+open Classical hiding choose axiomOfChoice
+
+/--
+warning: please avoid 'open (scoped) Classical' statements: this can hide theorem statements
+which would be better stated with explicit decidability statements.
+Instead, use `open Classical in` for definitions or instances, the `classical` tactic for proofs.
+For theorem statements, either add missing decidability assumptions or use `open Classical in`.
+note: this linter can be disabled with `set_option linter.openClassical false`
+-/
+#guard_msgs in
+open Classical renaming choose -> foo, byCases -> bar
+
+-- Only opening specific items.
+/--
+warning: please avoid 'open (scoped) Classical' statements: this can hide theorem statements
+which would be better stated with explicit decidability statements.
+Instead, use `open Classical in` for definitions or instances, the `classical` tactic for proofs.
+For theorem statements, either add missing decidability assumptions or use `open Classical in`.
+note: this linter can be disabled with `set_option linter.openClassical false`
+-/
+#guard_msgs in
+open Classical (choose)
+
+-- `open scoped Classical` is also linted
+
+/--
+warning: please avoid 'open (scoped) Classical' statements: this can hide theorem statements
+which would be better stated with explicit decidability statements.
+Instead, use `open Classical in` for definitions or instances, the `classical` tactic for proofs.
+For theorem statements, either add missing decidability assumptions or use `open Classical in`.
+note: this linter can be disabled with `set_option linter.openClassical false`
+-/
+#guard_msgs in
+open scoped Classical
+
```

Comment:
```suggestion

```
This one might also be superfluous, then!

Reply:
Well, this and the next test verify slightly different things: this one has only a single namespace, the other is about multiple of them. If you think this difference is really inessential, I can remove this one.

Reply:
No, it's ok to keep it.  When I looked at it, I think that GitHub was still showing the test that you had meanwhile removed.